core/server/models/client.js

```diff
@@ -16,9 +16,7 @@ Client = ghostBookshelf.Model.extend({
 
         return {
             uuid: uuid.v4(),
-            secret: secret,
-            status: 'development',
-            type: 'ua'
```

Comment:
I'm not entirely sure about these changes.

Bookshelf has a concept of saved & unsaved models. The reason for including the defaults in the model, is so that an unsaved model has the default values before being sent to the DB - so that the values can be validated and are present for any logic that happens.

I do not know for sure if we have cases like this, but I think this reasoning is worth considering?

Reply:
Thanks for the hint with unsaved models. Will double check 👍

Reply:
So `defaults` are called during the `save` call in bookshelf.
[See](https://github.com/bookshelf/bookshelf/blob/0.10.4/src/model.js#L941) bookshelf code section.

They are then available in the event hooks, **not earlier**. Validation doesn't have to happen for defaults - these are (mostly static) values, which we define. But if there is any connected logic it would be helpful to have access to them when *inserting* resources. Right now there is no connected logic, but i will still revert and leave a clear comment. We have to live with the fact that we have to define defaults twice (once in the schema, once in the model layer) - for now.